dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/task/AbstractJdbcTask.java

```diff
@@ -76,28 +83,65 @@ public AbstractJdbcTask<T> withHeaderClass(@Nonnull Class<? extends Enum<?>> hea
     return this;
   }
 
+  @Nonnull
+  public AbstractJdbcTask<T> withAssessment(boolean flag) {
+    this.isAssessment = flag;
+    return this;
+  }
+
+  @Nonnull
+  public AbstractJdbcTask<T> withHeaderValidation(boolean isRequired) {
+    this.requiresHeaderValidation = isRequired;
+    return this;
+  }
+
   @Nonnull
   protected CSVFormat newCsvFormat(@Nonnull ResultSet rs) throws SQLException {
     CSVFormat format = FORMAT;
     Class<? extends Enum<?>> headerClass = getHeaderClass();
     if (headerClass != null) {
       format = format.withHeader(headerClass);
-      if (headerClass.getEnumConstants().length != rs.getMetaData().getColumnCount())
-        // Can we avoid nesting exceptions here?
-        throw new SQLException(
-            new MetadataDumperUsageException(
-                "Fatal Error. ResultSet does not have the expected column count: "
-                    + headerClass.getEnumConstants().length,
-                Arrays.asList(
-                    "If a custom query has been specified please confirm the selected columns match"
-                        + " the following: ",
-                    StringUtils.join(headerClass.getEnumConstants(), ", "))));
+      ResultSetMetaData metaData = rs.getMetaData();
+      validateHeader(headerClass, metaData);
     } else {
       format = format.withHeader(rs);
     }
     return format;
   }
 
+  private void validateHeader(Class<? extends Enum<?>> headerClass, ResultSetMetaData metaData)
+      throws SQLException {
+    if (isAssessment) {
```

Comment:
Can we please spend a few minutes talking about why this should be an assessment-specific behaviour? I'm keen to keep the dumper as a single product, particularly when we have to load the dumps into two downstream consumers: translation and assessment. Also, we have just received a report of invalid SQL generated by the dumper, causing task failure, and the bifurcation of the SQL generation is making it harder to debug and test.

Reply:
Changed the flag to `ignoreExtraColumns`